src/compression.rs

```diff
@@ -1236,9 +1232,10 @@ fn get_refinements_of_shifted_id(shifted_id: Id, egraph: &crate::EGraph, cfg: &C
 /// figure out all the N^2 zippers from choosing any given node and then choosing a descendant and returning the zipper from
 /// the node to the descendant. We also collect a bunch of other useful stuff like the argument you would get if you abstracted
 /// the descendant and introduced an invention rooted at the ancestor node.
+#[allow(clippy::type_complexity)]
```

Comment:
clippy doesn't like this return type. I don't blame it. I've disabled the warning for now, but we should probably consider giving this type a more legible/explicit definition.